crates/wasm/src/wasm_planner.rs

```diff
@@ -0,0 +1,225 @@
+use indexed_db_futures::prelude::OpenDbRequest;
+use indexed_db_futures::{IdbDatabase, IdbQuerySource};
+use penumbra_dex::swap_claim::SwapClaimPlan;
+use penumbra_proto::core::chain::v1alpha1::{ChainParameters, FmdParameters};
+use penumbra_proto::core::crypto::v1alpha1::{Address, DenomMetadata, Fee, StateCommitment, Value};
+use penumbra_proto::core::transaction::v1alpha1::{MemoPlaintext, TransactionPlan};
+use penumbra_proto::view::v1alpha1::NotesRequest;
+use penumbra_proto::DomainType;
+use rand_core::OsRng;
+use wasm_bindgen::prelude::wasm_bindgen;
+use wasm_bindgen::JsValue;
+
+#[wasm_bindgen]
+pub struct WasmPlanner {
+    planner: Planner<OsRng>,
+}
+
+#[wasm_bindgen]
+impl WasmPlanner {
+    #[wasm_bindgen(constructor)]
+    pub fn new() -> WasmPlanner {
+        WasmPlanner {
+            planner: Planner::new(OsRng),
+        }
+    }
+
+    pub fn expiry_height(&mut self, expiry_height: JsValue) -> Result<(), JsValue> {
+        self.planner
+            .expiry_height(serde_wasm_bindgen::from_value(expiry_height)?);
+
+        Ok(())
+    }
+
+    pub fn memo(&mut self, memo: JsValue) -> Result<(), JsValue> {
+        let memo_proto: MemoPlaintext = serde_wasm_bindgen::from_value(memo)?;
+
+        let _ = self.planner.memo(memo_proto.try_into().unwrap());
+
+        Ok(())
+    }
+
+    pub fn fee(&mut self, fee: JsValue) -> Result<(), JsValue> {
+        let fee_proto: Fee = serde_wasm_bindgen::from_value(fee)?;
+
+        self.planner.fee(fee_proto.try_into().unwrap());
+
+        Ok(())
+    }
+
+    pub fn output(&mut self, value: JsValue, address: JsValue) -> Result<(), JsValue> {
+        let value_proto: Value = serde_wasm_bindgen::from_value(value)?;
+        let address_proto: Address = serde_wasm_bindgen::from_value(address)?;
+
+        self.planner.output(
+            value_proto.try_into().unwrap(),
+            address_proto.try_into().unwrap(),
+        );
+
+        Ok(())
+    }
+
+    pub async fn swap_claim(&mut self, swap_commitment: JsValue) -> Result<(), JsValue> {
+        utils::set_panic_hook();
+
+        let swap_commitment_proto: StateCommitment =
+            serde_wasm_bindgen::from_value(swap_commitment)?;
+
+        let swap_record = get_swap_by_commitment(swap_commitment_proto).await.unwrap();
+        let chain_params_proto: ChainParameters = get_chain_parameters().await.unwrap();
+
+        let swap_claim_plan = SwapClaimPlan {
+            swap_plaintext: swap_record.swap,
+            position: swap_record.position,
+            output_data: swap_record.output_data,
+            epoch_duration: chain_params_proto.epoch_duration,
+            proof_blinding_r: Fq::rand(&mut OsRng),
+            proof_blinding_s: Fq::rand(&mut OsRng),
+        };
+
+        self.planner.swap_claim(swap_claim_plan);
+        Ok(())
+    }
+
+    pub fn swap(
+        &mut self,
+        input_value: JsValue,
+        into_denom: JsValue,
+        swap_claim_fee: JsValue,
+        claim_address: JsValue,
```

Comment:
Can't these be?:

```rust
input_value: u128,
into_denom: &str,
swap_claim_fee: u128,
claim_address: &str,
```

Whenever possible, we should not do JsValue if it is allowed. You may have to test this.

Reply:
Same feedback as before 👆

Reply:
The point is that on the TS side we will be using mostly [proto](https://buf.build/penumbra-zone/penumbra) types to interact with wasm. 
And in this case, having an object of the [Address](https://buf.build/penumbra-zone/penumbra/docs/main:penumbra.core.crypto.v1alpha1#penumbra.core.crypto.v1alpha1.Address) type on the TS side, we will need to 
1. convert Address -> string on the TS side 
2. on the wasm side get an Address from string object
This leads to additional writing of specific conversion code, while the use of JsValue will be unified everywhere else